packages/admin/admin-generator/src/commands/generate/generateGrid/generateGrid.ts

```diff
@@ -405,8 +406,14 @@ export function generateGrid<T extends { __typename?: string }>(
 
         if (type == "dateTime") {
             gridColumnType = "...dataGridDateTimeColumn,";
+            valueGetter = name.includes(".")
+                ? `(params, row) => row.${name.replace(/\./g, "?.")} && new Date(row.${name.replace(/\./g, "?.")})`
+                : undefined;
```

Comment:
this is working, because of the [default valueGetter implementation in dataGridDateTimeColumn](https://github.com/vivid-planet/comet/blob/main/packages/admin/admin/src/dataGrid/gridColumnTypes.tsx#L109):
`valueGetter: (value) => value && new Date(value),`

by setting it to undefined the default is used.

(you could have explain that, it took me quite some time to figure all that out)

Alternative Idea: we could move `new Date` into dataGridDateTimeColumn.renderCell or use valueFormatter? (does filtering & sorting (client side) break with this? what about export?)

Reply:
This PR is only a fix. It does the same as `dataGridDateTimeColumn` and `dataGridDateColumn`, only using the nested field.

`valueGetter: (value) => value && new Date(value)` vs `valueGetter: (value, row) => row.manufacturer.createAt && new Date(row.manufacturer.createAt)`.

So discussion about the generated code is probably not required to complete this PR? ;-)

But I'm okay to discuss if we should change how we handle date-columns in a follow up PR. (But I currently don't have an opinion how to solve this differently...)

Reply:
I think this would cause some issues. 
The value used by `renderCell` comes from the `valueGetter` which expects a `Date` (`valueGetter` ensures it's actually a `Date`). 
It's also used by `generateExcelFile()`, I think to determine which type of cell to render, e.g., if a `Date` is passed in, it will be a date-cell (shown in the Excel-users locale) instead of a string with the date as formatted by the database.

Reply:
There are 3 relevant functions:

- render in grid: `row[field]` -> `valueGetter` -> `valueFormatter` -> `renderCell`
- our `generateExeclFile()`: `row[field]` -> `valueGetter` -> `valueFormatter`
- filtering: `row[field]` -> `valueGetter`
- sorting: `row[field]` -> `valueGetter`

If we would move the string-to-date-object conversion into `valueFormatter`, we have it in render-in-grid and excel-export as Date object, but not in filtering and sorting. Sorting by string will work for sure; for filtering we could implement a custom `getApplyFilterFn` (defined in `dataGridDateTimeColumn`) that supports strings. However: in most of our cases we do remote filtering anyway.

Reply:
> If we would move the string-to-date-object conversion into valueFormatter

Is this a good idea?

> Default methods, such as filtering and sorting, assume that the type of the values will match the type of the column specified in type. For example, values of column with type: 'dateTime' are expecting to be stored as a Date() objects. If for any reason, your data type is not the correct one, you can use valueGetter to parse the value to the correct type.
> – https://mui.com/x/react-data-grid/column-definition/#converting-types

if we were to remove `valueGetter` from our column helpers, we need to either change the column type to something other than 'dateTime' or always call valueGetter in the generated code.